client/components/theme/index.jsx

```diff
@@ -347,27 +351,37 @@ export class Theme extends Component {
 		}
 
 		return (
-			<ThemeCard
-				ref={ this.props.bookmarkRef }
-				name={ name }
-				description={ themeDescription }
-				image={ this.renderScreenshot() }
-				imageClickUrl={ this.props.screenshotClickUrl }
-				imageActionLabel={ this.props.actionLabel }
-				banner={ this.renderUpdateAlert() }
-				badge={ this.renderBadge() }
-				styleVariations={ style_variations }
-				selectedStyleVariation={ selectedStyleVariation }
-				optionsMenu={ this.renderMoreButton() }
-				isActive={ this.props.active }
-				isLoading={ this.props.loading }
-				isSoftLaunched={ this.props.softLaunched }
-				isShowDescriptionOnImageHover={ ! isCustomGeneratedTheme }
-				onClick={ this.setBookmark }
-				onImageClick={ this.onScreenshotClick }
-				onStyleVariationClick={ this.onStyleVariationClick }
-				onStyleVariationMoreClick={ this.onStyleVariationClick }
-			/>
+			<InView triggerOnce rootMargin={ VIEWPORT_HEIGHT }>
+				{ ( { inView, ref } ) => (
+					<div ref={ ref }>
+						{ inView ? (
+							<ThemeCard
+								ref={ this.props.bookmarkRef }
+								name={ name }
+								description={ themeDescription }
+								image={ this.renderScreenshot() }
+								imageClickUrl={ this.props.screenshotClickUrl }
+								imageActionLabel={ this.props.actionLabel }
+								banner={ this.renderUpdateAlert() }
+								badge={ this.renderBadge() }
+								styleVariations={ style_variations }
+								selectedStyleVariation={ selectedStyleVariation }
+								optionsMenu={ this.renderMoreButton() }
+								isActive={ this.props.active }
+								isLoading={ this.props.loading }
+								isSoftLaunched={ this.props.softLaunched }
+								isShowDescriptionOnImageHover={ ! isCustomGeneratedTheme }
+								onClick={ this.setBookmark }
+								onImageClick={ this.onScreenshotClick }
+								onStyleVariationClick={ this.onStyleVariationClick }
+								onStyleVariationMoreClick={ this.onStyleVariationClick }
+							/>
+						) : (
+							this.renderPlaceholder()
+						) }
+					</div>
+				) }
+			</InView>
```

Comment:
What do you think of spreading the props so that we can have a more concise component? Something like:
```
<InView triggerOnce rootMargin={ VIEWPORT_HEIGHT }>
	{ ( { inView, ref } ) => (
		<div ref={ ref }>
			{ inView ? (
				<ThemeCard
					ref={ this.props.bookmarkRef }
					{ ...this.props } // Spread props to reduce redundancy
					image={ this.renderScreenshot() }
					banner={ this.renderUpdateAlert() }
					badge={ this.renderBadge() }
					optionsMenu={ this.renderMoreButton() }
					isShowDescriptionOnImageHover={ !isCustomGeneratedTheme }
					onClick={ this.memoizedSetBookmark }
					onImageClick={ this.memoizedScreenshotClick }
					onStyleVariationClick={ this.onStyleVariationClick }
					onStyleVariationMoreClick={ this.onStyleVariationClick }
				/>
			) : (
				this.renderPlaceholder()
			) }
		</div>
	)}
</InView>
```

Reply:
You mean renaming the `ThemeCard` props , right?

Reply:
Yes, it's less explicit but more concise. Don't need to change if you don't want to.

Reply:
I think I'll skip 😅